File: Hashing/01_subset_or_not.py
def is_subset_or_not(arr,arr2):
    newarr = set(arr)

    for i in range(len(arr2)):
        if arr2[i] not in newarr:
            return 0

    return 1



arr = [1,2,3,4,5,6]
arr2 = [2,3,4,1]
if is_subset_or_not(arr,arr2):
    print("SUBSET")
else:
    print("NOT SUBSET")

File: Hashing/01_subset_or_not.java

import java.util.ArrayList;
import java.util.HashSet;
import java.util.Set;

class SubsetOrNotCheck 
{

    public static boolean isSubsetOrNot(ArrayList<Integer> arr, ArrayList<Integer> arr2) 
    {
        Set<Integer> hashset = new HashSet<>();
        for (int num : arr) 
        {
            hashset.add(num);
        }

        for (int num : arr2) 
        {
            if (!hashset.contains(num)) 
            {
                return false;
            }
        }
        return true;

    }

    public static void main(String[] args) 
    {
        ArrayList<Integer> arr = new ArrayList<>();
        arr.add(1);
        arr.add(2);
        arr.add(3);
        arr.add(4);
        arr.add(5);
        ArrayList<Integer> arr2 = new ArrayList<>();
        arr2.add(2);
        arr2.add(1);
        arr2.add(4);
        arr2.add(5);
        if (isSubsetOrNot(arr, arr2)) {
            System.out.println("SUBSET"); 
        }else {
            System.out.println("NOT subset");
        }
    }
}
